### PR TITLE
Exclude Lambda Extension spans when using tracer

### PR DESCRIPTION
**Pull request by nhinsch**
### What does this PR do?

Configures the tracer's HTTP plugin to exclude spans representing HTTP calls from the Lambda Library to the Lambda Extension (by checking the URL for the port `:8124`).

Note that this will only work if the customer is using the provided handler function and the handler redirection feature, because the handler is where the library initializes the tracer. If the customer instead initializes the tracer in their own code, they would need to apply this configuration there as well.

Bonus: fix the `publish_sandbox` script and disable X-Ray in the integration tests.

### Motivation

By default, the tracer will create a span when the Lambda Library calls the `/flush` endpoint. Because the `/flush` HTTP call from the Library to the Extension occurs outside of the function invocation, it is created as an entirely separate trace from the function invocation. These extraneous traces are potentially confusing and unhelpful.

### Testing Guidelines

I tested manually. Integration tests for the Extension will also help test this behavior.

### Types of Changes

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: integration_tests/serverless.yml [hunk 1]
```diff
@@ -4,9 +4,6 @@ service: integration-tests-js
 provider:
   name: aws
   region: sa-east-1
-  tracing:
```
Comment (nhinsch) [inline integration_tests/serverless.yml:7]: I disabled X-Ray tracing in the integration tests. Due to a bug on the AWS side, the `XRAY` log lines are now improperly formatted.

File: src/handler.ts [hunk 2]
```diff
@@ -15,12 +15,19 @@ if (getEnvValue("DD_TRACE_ENABLED", "true").toLowerCase() === "true") {
...
     tags: {
       "_dd.origin": "lambda",
     },
   });
   logDebug("automatically initialized dd-trace");
+
+  // Configure the tracer to ignore HTTP calls made from the Lambda Library to the Extension
+  tracer.use("http", {
+    // blacklist has been deprecated in favor of blocklist, but we still use it here for
+    // compatibility with older versions of the tracer
```
Comment (hghotra) [inline src/handler.ts:28]: 👍🏼

File: integration_tests/serverless.yml (hunk 1 above)
Comment (DarcyRaynerDD) [reply integration_tests/serverless.yml:7]: Is this the same issue we were seeing with Ruby?

Comment (nhinsch) [reply integration_tests/serverless.yml:7]: Sounds like it

**Pull request merged**